rust/worker/src/execution/operators/merge_metadata_results.rs

```diff
@@ -379,15 +391,19 @@ mod test {
         storage::{local::LocalStorage, Storage},
         types::{LogRecord, MetadataValue, Operation, OperationRecord, UpdateMetadataValue},
     };
-    use std::{collections::HashMap, str::FromStr};
-    use uuid::Uuid;
 
     #[tokio::test]
     async fn test_merge_and_hydrate() {
         let tmp_dir = tempfile::tempdir().unwrap();
         let storage = Storage::Local(LocalStorage::new(tmp_dir.path().to_str().unwrap()));
-        let arrow_blockfile_provider =
-            ArrowBlockfileProvider::new(storage, TEST_MAX_BLOCK_SIZE_BYTES);
+        let block_cache = Cache::new(&CacheConfig::Unbounded(UnboundedCacheConfig {}));
```

Comment:
I see - I presume test configuration is why we promote a bunch of types. Could we for cleanliness keep the types inside their respective modules and use configuration to allow for transparent creation?